Tests/NIOKitTests/FutureExtensionsTests.swift

```diff
@@ -0,0 +1,34 @@
+import XCTest
+import NIO
+@testable import NIOKit
+
+final class FutureExtensionsTests: XCTestCase {
+	private var group: EventLoopGroup!
+	private var eventLoop: EventLoop { return group.next() }
+    
+    override func setUp() {
+        group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+	}
+    
+    override func tearDown() {
+        XCTAssertNoThrow(try group.syncShutdownGracefully())
+        group = nil
+	}
+	
+	func testGuard() {
+		let future1 = eventLoop.makeSucceededFuture(1)
+		let guardedFuture1 = future1.guard({ $0 == 1 }, else: TestError.notEqualTo1)
+		XCTAssertNoThrow(try guardedFuture1.wait())
+		
+		let future2 = eventLoop.makeSucceededFuture("foo")
+		let guardedFuture2 = future2.guard({ $0 == "bar" }, else: TestError.notEqualToBar)
+		XCTAssertThrowsError(try guardedFuture2.wait())
+	}
+	
+	static var allTests = [("testGuard", testGuard)]
```

Comment:
each tuple should be on its own line with commas after (to minimize merge conflicts)

Reply:
Comma even if only one like `("testGuard", testGuard),`?

Reply:
Yup... separating each element (including the brackets) to its own line makes for a cleaner merge situation:

```swift
static var allTests = [
    ("testGuard", testGuard),
]
```